src/scenic/core/scenarios.py

```diff
@@ -1,518 +1,595 @@
+                 temporalReqs=(),terminationConds=(), termSimulationConds=(),
+                 recordedExprs=(), recordedInitialExprs=(), recordedFinalExprs=(),
+                 monitors=(), behaviorNamespaces={}, dynamicScenario=None,
+                 sample={}, compileOptions={}):
+        self.workspace = workspace
+        self.objects = tuple(objects)
+        self.egoObject = egoObject
+        self.params = params
+        self.temporalRequirements = tuple(temporalReqs)
+        self.terminationConditions = tuple(terminationConds)
+        self.terminateSimulationConditions = tuple(termSimulationConds)
+        self.recordedExprs = tuple(recordedExprs)
+        self.recordedInitialExprs = tuple(recordedInitialExprs)
+        self.recordedFinalExprs = tuple(recordedFinalExprs)
+        self.monitors = tuple(monitors)
+        self.behaviorNamespaces = behaviorNamespaces
+        self.dynamicScenario = dynamicScenario
+        self.sample = sample
+        self.compileOptions = compileOptions
+
+    def dumpAsScenicCode(self, stream=sys.stdout):
+        """Dump Scenic code reproducing this scene to the given stream.
+
+        For non-human-readable but complete serialization of scenes see
+        `Scenario.sceneToBytes` and `Scenario.sceneFromBytes`.
+
+        .. note::
+
+            This function does not currently reproduce parts of the original Scenic
+            program defining behaviors, functions, etc. used in the scene. Also, if
+            the scene involves any user-defined types, they must provide a suitable
+            :obj:`~object.__repr__` for this function to print them properly.
+
+        Args:
+            stream (:term:`text file`): Where to print the code (default `sys.stdout`).
+        """
+        for name, value in self.params.items():
+            if str.isidentifier(name):
+                stream.write(f'param {name} = ')
+            else:
+                stream.write(f'param "{name}" = ')
+            dumpAsScenicCode(value, stream)
+            stream.write('\n')
+        stream.write('ego = ')
+        for obj in self.objects:
+            dumpAsScenicCode(obj, stream)
+            stream.write('\n')
+
+    def show(self, zoom=None, block=True):
+        self.show3D()
+
+    def show3D(self):
+        """Render a 3D schematic of the scene for debugging."""
+        import trimesh
+
+        # Create a new trimesh scene to contain meshes
+        render_scene = trimesh.scene.Scene()
+
+        # display map
+        self.workspace.show3D(render_scene)
+        # draw objects
+        for obj in self.objects:
+            obj.show3D(render_scene, highlight=(obj is self.egoObject))
+
+        # If nothing else is in the viewer, add some constructs
+        # to avoid a crash
+        if render_scene.is_empty:
+            render_scene.add_geometry(trimesh.points.PointCloud([(0.1,0,0),(0,0.1,0), (0,0,.1)], colors=[255,255,255,0]))
+
+        render_scene.show(flags={'axis': 'world'})
```

Comment:
Would be nice to have an option for disabling the axis indicator.

Reply:
It's now off by default and triggered with the `--axis` flag.